### ISSUE TITLE
Grade Summary Order Non-Deterministic

### ISSUE THREAD

**Issue by holzbh** [Submitty/Submitty#1381]
After doing several `vagrant destroy ubuntu ; vagrant up ubuntu --provision`, letting grading finish (requires a restart of the autograder still), and then doing Generate Grade Summaries, the resulting summaries for `sample` are in a different order each time. The Rainbow Grades test expects summaries in the same order each time, so either we need the ordering to be deterministic or we need to relax that requirement.

The advantage of checking the summary files is that if the test fails / behavior changes, it's valuable to know the expected input has changed vs the behavior/output of Rainbow Grades itself.

**Comment by holzbh** [comment]
Needs further investigation, on a branch that was running master up through f86d22905a68e46bac85a3c655891a9b0043db39, each time I did a destroy and up I got a different output.html with varying grades. I've attached a .zip file with the three output.html results I got.

Running partial_reset does produce the same output.html (as long as I don't destroy and up again).

[issue1381_outputs.zip](https://github.com/Submitty/Submitty/files/1253409/issue1381_outputs.zip)
